contributors/design-proposals/support-union-in-api-server.md

```diff
@@ -0,0 +1,200 @@
+Add Tag Indicating To Replace *Union* Fields in Patch
+=============
+
+Add a tag `patchStrategy:"replaceKey"`. For a given type that has this tag, all keys/fields missing 
+from the request will be cleared when patching the object. For a field presents in the request,
+it will be merged with the live config.
+
+This approach could be used to fix the issues with `kubectl apply` (e.g. [#34292](https://github.com/kubernetes/kubernetes/issues/34292))
+
+The proposal of Full Union is in [kubernetes/community#388](https://github.com/kubernetes/community/pull/388).
+
+| Capability | Supported By This Proposal | Supported By Full Union |
+|---|---|---|---|
+| Auto clear missing fields on patch | X | X |
+| Merge union fields on patch | X | X |
+| Validate only 1 field set on type |  | X |
+| Validate discriminator field matches one-of field |  | X |
+| Support non-union patchKey | X | TBD |
+| Support arbitrary combinations of set fields | X |  |
+
+## Proposed Changes
+
+### APIs
+
+**Scope**:
+
+| Union Type | Supported |
+|---|---|---|
+| non-inlined non-discriminated union | Yes |
+| non-inlined discriminated union | Yes |
+| inlined union with [patchMergeKey](https://github.com/kubernetes/community/blob/master/contributors/devel/api-conventions.md#strategic-merge-patch) only | Yes |
+| other inlined union | No |
+
+For the *inlined* fields, we have 2 options:
+
+1) Move the tag to the parent struct's.
+
+2) Add some logic to lookup the metadata in go struct of inline union.
+
+We are leaning more to option 1, because the limitation of option 2 is that the metadata associated with
+the inlined APIs will not be reflected in the OpenAPI schema.
+
+#### Examples
+
+There are some examples using option 1 below.
+
+Example of non-inlined non-discriminated union:
+```go
+type ContainerStatus struct {
```

Comment:
Include for each of these examples the actual contents of the: live object + merge patch + result.  Also include examples with apply file configuration + live object + result.